### ISSUE TITLE
[Bug]: Cannot create TLS server when running from Electron. Works fine with node.

### ISSUE THREAD

**Issue by kowalski** [electron/electron#31042]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

14.0.1

### What operating system are you using?

Ubuntu

### Operating System Version

Linux 5.8.0-59-generic

### What arch are you using?

x64

### Expected Behavior

It should be possible to create ssl server using `https.createServer` passing valid certificates.

I've created a minimal repository reproducing the problem here: https://github.com/rumblefishdev/electron-ssl-problem

I can create a server when running `index.js` from node just fine. I've also tested my certificates using `openssl s_client` and it works just fine.

### Actual Behavior

When running from electron I'm getting an error:
```
Error: error:1000012e:SSL routines:OPENSSL_internal:KEY_USAGE_BIT_INCORRECT
```
coming from internals of TLS.

I've tested this on Electron versions: 14.0.1 (latest stable) and 12.0.2. 

### Testcase Gist URL

https://github.com/rumblefishdev/electron-ssl-problem

**Comment by BlackHole1** [comment]
This should be the behavior of chromium, see : https://bugs.chromium.org/p/chromium/issues/detail?id=795089

**Comment by BlackHole1** [comment]
The problem is that your certificate is missing `keyUsage`
You can use https://gist.github.com/sethvargo/81227d2316207b7bd110df328d83fad8 to generate your certificate

**Comment by kowalski** [comment]
@BlackHole1 thank you for your response. You're obviously right saying that BoringSSL doesn't like my KeyUsage field. This is how my certificate looks like when viewed using openssl:

```
$ openssl x509 -in a.pem -text -noout
Certificate:
    Data:
        Version: 1 (0x0)
        Serial Number: 1632222075441 (0x17c08045a31)
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: CN = Zbay CA
        Validity
            Not Before: Dec 28 10:10:10 2010 GMT
            Not After : Dec 28 10:10:10 2030 GMT
        Subject: CN = wfshemkk2k7mhimmdsck37atfzks2if62msd33ozbkh6pozjecevxzad.onion
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:00:74:bf:1d:cf:19:69:be:fc:d0:87:a2:8f:33:
                    64:a8:f8:9b:47:be:c9:84:e9:42:33:2d:70:5a:95:
                    c6:8b:ef:6b:92:09:81:14:25:09:6e:43:c1:b7:a8:
                    6a:45:74:60:ee:1a:f9:84:c9:40:4f:7a:d6:64:95:
                    2b:2b:da:2f:df
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            X509v3 Key Usage: 
                Key Agreement, Certificate Sign, CRL Sign
            X509v3 Extended Key Usage: 
                TLS Web Client Authentication, TLS Web Server Authentication
            1.2.840.113549.1.9.12: 
                . ..<.....M........0.....)x...B...
            1.3.6.1.4.1.50715.2.1: 
                ..fffffff
            1.3.6.1.2.1.15.3.1.1: 
                ..QmYqhQJ2fa2QoifTKNDNozqktdvotartW8J68F6bRHDJJn
    Signature Algorithm: ecdsa-with-SHA256
         30:44:02:20:39:42:78:13:f1:f6:1d:42:2e:cc:a7:a8:55:07:
         1a:a4:b3:d0:08:41:85:51:b0:49:2f:00:ae:52:52:2f:e3:5a:
         02:20:77:1d:15:69:34:6a:ea:63:a9:3b:03:80:06:10:96:9f:
         6d:07:8a:a9:7e:41:79:bc:9f:db:cf:69:b4:43:2c:6b
```

So I do have: Key Agreement, Certificate Sign, CRL Sign, but I don't have Digital Signature, which I think BoringSSL requires.

In any case, the fix for me is to figure flip additional bit, just figure out which one.

However I still think that this is a (minor) bug in Electron/BoringSSL. The certificate I have works fine in OpenSSL, both when using `s_server` and TLS module from nodejs. I think it's wrong for Electron to deviate from standards set by OpenSSL. 

I can work this problem around, because I control how the certificate is generate. In the scenario one is given the certificate generated by third party, it's not going to be usable in Electron.

Thank you for your help.

**Comment by BlackHole1** [comment]
This is the expected behavior of `BoringSSL`, see:

https://github.com/google/boringssl/blob/18b6836b2f6340187a7981c82be9be9d092d36d6/ssl/handshake_server.cc#L1576-L1578

Because electron uses chromium, it must use BoringSSL instead of openSSL, see: https://github.com/electron/electron/issues/20204#issuecomment-530820074

**Comment by kowalski** [comment]
I guess Chromium is not usually used to start TLS server, they may not be aware that they deviated from OpenSSL.
Obviously Chromium is not going to change this, it would be hard to argue that this is causing any problem in their product.

What I thought could be done on Electron's end is adding another patch here: https://github.com/electron/electron/tree/main/patches/boringssl to make the behavior consistent with OpenSSL.

**Comment by BlackHole1** [comment]
Yes, but the problem is that electron is very strict about patch behavior

As I said before, electron will not consider this unless there is a very good reason to do so

cc @nornagon @codebytere What about your suggestions?

**Comment by codebytere** [comment]
Generally speaking we add BoringSSL patches when compilation failures occur as a result of a Node.js upgrade or when a nontrivial number of users demonstrate a need - however, I can reach out to BoringSSL and see if we can come to a mutually agreeable solution and report back.
